pkg/sql/tenant.go

```diff
@@ -185,8 +185,35 @@ func CreateTenantRecord(
 	)
 }
 
-// GetTenantRecord retrieves a tenant in system.tenants.
-func GetTenantRecord(
+// GetTenantRecordByName retrieves a tenant with the provided name from
+// system.tenants.
+func GetTenantRecordByName(
+	ctx context.Context, execCfg *ExecutorConfig, txn *kv.Txn, tenantName roachpb.TenantName,
+) (*descpb.TenantInfo, error) {
+	if !execCfg.Settings.Version.IsActive(ctx, clusterversion.V23_1TenantNames) {
+		return nil, errors.Newf("tenant names not supported until upgrade to %s or higher is completed",
+			clusterversion.V23_1TenantNames.String())
+	}
+	row, err := execCfg.InternalExecutor.QueryRowEx(
+		ctx, "activate-tenant", txn, sessiondata.NodeUserSessionDataOverride,
+		`SELECT info FROM system.tenants WHERE name = $1`, tenantName,
+	)
+	if err != nil {
+		return nil, err
+	} else if row == nil {
+		return nil, pgerror.Newf(pgcode.UndefinedObject, "tenant \"%q\" does not exist", tenantName)
+	}
+
+	info := &descpb.TenantInfo{}
+	infoBytes := []byte(tree.MustBeDBytes(row[0]))
+	if err := protoutil.Unmarshal(infoBytes, info); err != nil {
+		return nil, err
+	}
+	return info, nil
+}
+
+// GetTenantRecordByID retrieves a tenant in system.tenants.
+func GetTenantRecordByID(
 	ctx context.Context, execCfg *ExecutorConfig, txn *kv.Txn, tenID uint64,
```

Comment:
It's existing code so don't feel obliged to change it. But it seems odd to take a uint64 here rather than a TenantID.

Reply:
good point, fixed.